src/metatensor/models/utils/neighbors_lists.py

```diff
@@ -35,9 +35,10 @@ def attach_neighbor_lists(
 
     # Compute the neighbor lists
     for options in neighbor_lists:
-        neighbor_list = _compute_single_neighbor_list(atoms, options)
-        register_autograd_neighbors(system, neighbor_list)
-        system.add_neighbors_list(options, neighbor_list)
+        if options not in system.known_neighbors_lists():
```

Comment:
I'm a bit worried about this addition. Did this try to add a NL that was already there?

Reply:
Yes exactly, for some reason the double NL addition happens, and it causes an error

Reply:
Ok, some logic is breaking then but the change itself is harmless